src/metabase/models/serialization/utils.clj

```diff
@@ -0,0 +1,67 @@
+(ns metabase.models.serialization.utils
+  "Defines several helper functions and protocols for the serialization system.
+  Serialization is an enterprise feature, but in the interest of keeping all the code for an entity in one place, these
+  methods are defined here and implemented for all the exported models.
+
+  Whether to export a new model:
+  - Generally, the high-profile user facing things (databases, questions, dashboards, snippets, etc.) are exported.
+  - Internal or automatic things (users, activity logs, permissions) are not.
+
+  If the model is not exported, add it to the exclusion lists in the tests.
+
+  For models that are exported, you have to implement this file's protocols and multimethods for it:
+  - All exported models should either have an CHAR(21) column `entity_id`, or a portable external name (like a database
+    URL).
+  - identity-hash-fields should give the list of fields that distinguish an instance of this model from another, on a
+    best-effort basis.
+    - Use things like names, labels, or other stable identifying features.
+    - NO numeric database IDs!
+    - Any foreign keys should be hydrated and the identity-hash of the foreign entity used as part of the hash.
+      - There's a `hydrated-hash` helper for this with several example uses."
+  (:require [potemkin.types :as p.types]
+            [toucan.hydrate :refer [hydrate]]))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                              Identity Hashes                                                   |
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; Generated entity_id values have lately been added to most exported models, but they only get populated on newly
+;;; created entities. Since we can't rely on entity_id being present, we need a content-based definition of identity for
+;;; all exported models.
+(p.types/defprotocol+ IdentityHashable
+  (identity-hash-fields
+    [entity]
+    "You probably want to call metabase.models.serialization/identity-hash instead of calling this directly.
```

Comment:
nit: this is an incorrect reference. Also I think it's our convention to write a function name inside a double square bracket (e.g: [[metabase.models.user/User]])

This makes it easy to identify fn vs normal text, and on some IDE it has better syntax coloring for them